common/persistence/dataInterfaces.go

```diff
@@ -1064,7 +1090,13 @@ type (
 
 		// Tasks related APIs
 
+		// Hints for persistence implementaion regarding hisotry task readers
+		RegisterHistoryTaskReader(ctx context.Context, request *RegisterHistoryTaskReaderRequest) error
+		UnregisterHistoryTaskReader(ctx context.Context, request *UnregisterHistoryTaskReaderRequest) error
+		UpdateHistoryTaskReaderProgress(ctx context.Context, request *UpdateHistoryTaskReaderProgressRequest) error
```

Comment:
Regarding returned `error`. There's not much oss can to other than log the message I guess. 

For `RegisterHistoryTaskReader`, if error is returned, I can probably block the corresponding split operation. But I think underlying implementation can also rely on the fact that OSS will only register at most a fix number of readers (currently 2), so `error` is not really necessary here.

Reply:
Synced offline. Need to additional logic to handle the error from `RegisterHistoryTaskReader`. 

For Unregister and Update, there won't be any error returned so removed. We can add them back when later if needed and decide how to handle them.